alert-database/src/main/java/com/synopsys/integration/alert/database/api/DefaultProviderDataAccessor.java

```diff
@@ -169,4 +174,67 @@ private ProviderUserEntity convertToUserEntity(final String providerName, final
         return new ProviderUserEntity(providerUserModel.getEmailAddress(), providerUserModel.getOptOut(), providerName);
     }
 
+    @Override
+    public void updateProjectAndUserData(final String providerName, final Map<ProviderProject, Set<String>> projectToUserData) {
+        updateProjectDB(providerName, projectToUserData.keySet());
+        final Set<String> userData = projectToUserData.values().stream().flatMap(userDataSet -> userDataSet.stream()).collect(Collectors.toSet());
+        updateUserDB(providerName, userData);
+        updateUserProjectRelations(projectToUserData);
+    }
+
+    private List<ProviderProject> updateProjectDB(final String providerName, final Set<ProviderProject> currentProjects) {
+        logger.info("{} projects", currentProjects.size());
+        return deleteAndSaveAllProjects(providerName, currentProjects);
```

Comment:
Considering Polaris and Black Duck will both depend on this functionality eventually, we should probably track the deltas like we do with users.

Reply:
Created a ticket to update this